core/src/test/java/hudson/model/UserTest.java

```diff
@@ -43,4 +43,31 @@ public void blankIdsOrFullNamesShouldNotBeAllowed() {
         assertThat("Blank user IDs should not be allowed", User.isIdOrFullnameAllowed("      "), is(false));
     }
     
+    @Test
+    @Issue("JENKINS-35967")
+    public void shoudNotAllowIllegalRestrictedNamesInWrongCase() {
+        assertIdOrFullNameNotAllowed("system");
+        assertIdOrFullNameNotAllowed("System");
+        assertIdOrFullNameNotAllowed("SYSTEM");
+        assertIdOrFullNameNotAllowed("syStem");
+        assertIdOrFullNameNotAllowed("sYstEm");
+    }
+    
+    @Test
+    @Issue("JENKINS-35967")
+    public void shoudNotAllowIllegalRestrictedNamesEvenIfTrimmed() {
+        for (String username : User.getIllegalPersistedUsernames()) {
+            assertIdOrFullNameNotAllowed(username);
+            assertIdOrFullNameNotAllowed(" " + username);
+            assertIdOrFullNameNotAllowed(username + " ");
+            assertIdOrFullNameNotAllowed("      " + username + "    ");
+            assertIdOrFullNameNotAllowed("\t" + username + "\t");  
```

Comment:
accents should be displayed, its no displayable characters (leading/trailing) that are the potential issue, as well as characters that look the same as one in the username but are not.

This is not a simple solution to solve.  You may find someone in the CS realm has solved it - but I am not a unicode string look-alikie algorithm guru....

It is perfectly valid for usernames to contain accents, signs (@ _ - . etc.), non western characters, spaces (yup on some systems, and potentially other things not mentioned here.  AD [seems to allow](https://msdn.microsoft.com/en-us/library/bb726984.aspx) pretty much anything with a few exceptions - it appears to allow leading/trailing whitespace according to the spec however in Server2012R2 these are stripped; possibly in the UI.)